docs/template_plugin/tests/functional/op_reference/random_uniform.cpp

```diff
@@ -0,0 +1,140 @@
+
+namespace reference_tests {
+namespace {
+
+struct RandomUniformParams {
+    RandomUniformParams(const std::vector<int64_t>& paramOutShape, const Tensor& paramMinValue, const Tensor& paramMaxValue, ngraph::element::Type paramOutType,
+                        int64_t paramGlobalSeed, int64_t paramOpSeed, const Tensor& paramExpected, const std::string& test_name)
+        : out_shape(paramOutShape),
+          min_val(paramMinValue),
+          max_val(paramMaxValue),
+          out_type(paramOutType),
+          global_seed(paramGlobalSeed),
+          op_seed(paramOpSeed),
+          expected(paramExpected),
+          test_case_name(test_name) {}
+    std::vector<int64_t> out_shape;
+    Tensor min_val;
+    Tensor max_val;
+    ngraph::element::Type out_type;
+    int64_t global_seed;
+    int64_t op_seed;
+    Tensor expected;
+    std::string test_case_name;
+};
+
+class ReferenceRandomUniformLayerTest : public testing::TestWithParam<RandomUniformParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params.out_shape, params.min_val, params.max_val, params.out_type, params.global_seed, params.op_seed);
+        inputData = {params.min_val.data, params.max_val.data};
+        refOutData = {params.expected.data};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<RandomUniformParams>& obj) {
+        auto param = obj.param;
+        return param.test_case_name;
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const std::vector<int64_t>& out_shape, const Tensor& min_val, const Tensor& max_val,
+                                                    ngraph::element::Type out_type, int64_t global_seed, int64_t op_seed) {
+        const auto min_val_param = std::make_shared<opset8::Parameter>(min_val.type, min_val.shape);
+        const auto max_val_param = std::make_shared<opset8::Parameter>(max_val.type, max_val.shape);
+        auto out_shape_ = std::make_shared<opset8::Constant>(element::i64, Shape {out_shape.size()}, out_shape);
+
+        return std::make_shared<Function>(
+            NodeVector {std::make_shared<opset8::RandomUniform>(out_shape_, min_val_param, max_val_param, out_type, global_seed, op_seed)},
+            ParameterVector {min_val_param, max_val_param});
+    }
+};
+
+TEST_P(ReferenceRandomUniformLayerTest, RandomUniformWithHardcodedRefs) {
+    Exec();
+}
+
+}  // namespace
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_RandomUniform_With_Hardcoded_Refs, ReferenceRandomUniformLayerTest,
+    ::testing::Values(
+        RandomUniformParams(
+            std::vector<int64_t> {3, 2, 4}, Tensor {{1}, element::f32, std::vector<float> {0}}, Tensor {{1}, element::f32, std::vector<float> {1}},
+            element::Type_t::f32, 150, 10,
+            Tensor {{3, 2, 4}, element::f32, std::vector<float> {0.7011235952377319336, 0.3053963184356689453, 0.9393105506896972656, 0.9456034898757934570,
```

Comment:
@ilyachur can we add tests with double tensors in template plugin? When I try to add double test I get error: Comparator for FP64 precision isn't supported

Reply:
At the current moment IE doesn't support FP64 precision